orangecontrib/spectroscopy/io/neaspec.py

```diff
@@ -310,4 +332,298 @@ def handle_data(self, data):
+        # its spacing interval will be calculated from metadata
+        # it will contain the number of points of the original data
+        # but with added padding on each side to accomodate the resampling
+        # to a common interval
+
+        if not self.info:
+            raise ValueError("No metadata found")
+        # parse info from the metadata to create the domain
+        interferometer_units, _, interferometer_distance = self.info[
+            "Interferometer Center/Distance"
+        ]
+        if interferometer_units != "[µm]":
+            raise ValueError("Interferometer units are not in micrometers")
+        interferometer_distance = (
+            float(interferometer_distance) * 1e-6
+        )  # convert [µm] to [m]
+        px_area_units, _, _, z = self.info["Pixel Area (X, Y, Z)"]
+        if px_area_units != "[px]":
+            raise ValueError("Pixel area units are not in pixels")
+        no_points = int(z)
+        m_min, m_max = self._define_M_range(df)
+        dm = self._domain_spacing(interferometer_distance, no_points)
+        # create the domain with the padding
+        d_start = m_min - padding * dm
+        d_end = m_min + (no_points + padding) * dm
+        if d_end < m_max:
+            raise ValueError(
+                "Could not create a domain with the given padding. \
+                             M maximum values are outside the expected range."
+            )
+        self.domain = np.arange(d_start, d_end, dm)
+        dx = 2 * float(dm) * 1e2  # convert [m] to [cm]
+        self.info["Calculated Datapoint Spacing (Δx)"] = ["[cm]", dx]
+
+    def create_original_df(self, fpath):
+        self.filename = fpath
+        self.info, header_length = self._read_table_metas(self.filename)
+        self.original_df = self._read_table_data(self.filename, header_length)
+        self.original_channel_names = self._extract_channel_names(self.original_df)
+
+    def create_cartesian_df(self):
+        if self.original_df.empty or not self.original_channel_names:
+            raise ValueError("Original data not found")
+
+        self.cartesian_df, self.cartesian_channel_names = self._create_cartesian_df(
+            self.original_df, self.original_channel_names
+        )
+        # clean up the original DataFrame
+        self.original_df = pd.DataFrame()
+
+    def create_resampled_df(self, padding=100):
+        if self.cartesian_df.empty or not self.cartesian_channel_names:
+            raise ValueError("Cartesian data not found")
+        if self.domain.size == 0:
+            raise ValueError("Domain not found")
+        self.resampled_df = self._create_resampled_polar_df_from_cartesian(
+            self.cartesian_df,
+            self.cartesian_channel_names,
+            self.domain,
+            padding=padding,
+        )
+        # clean up the cartesian DataFrame
+        self.cartesian_df = pd.DataFrame()
+
+    def read(self):
```

Comment:
Are you re-implementing `build_spec_table` (and not subclassing `SpectralFileFormat`) for this reader just to change the number of decimals in the feature names? Just curious.

Reply:
Hi, @stuart-cls ! Thanks for reviewing. 

Before re-implementing build_spec_table, I bumped on an error related to finding the string representation of some value, which seemed connected to domain creation. While I wasn’t completely sure, I agree that it might be related to the number of decimals. However, I couldn’t see a straightforward way to resolve this using build_spec_table.

The other 'Nea' readers use the array index as the data domain, but in this file, we have actual positions, as @raul-freitas mentioned. The scale of these positions could indeed be causing the issue. I considered changing the scale (e.g., from meters to micrometers), but this would obscure that information from the user, even if we added it to the metadata. Do you think adjusting the scale would be a better approach?

Reply:
Regarding the scale, I don't know what SNOM users expect but I imagine that micrometers would be more readable. Certainly in the OPUS/Agilent readers we use micrometers for map coordinates.

Reply:
Hello @brnovasco and @stuart-cls, thanks for advancing this. About the scale, microns is an appropriate unit for the M column. This column has the position readings of the stage encoder in microns.

Reply:
I believe the issue with scaling the domain from meters to micrometers is related to the formatting of the values used as column names in the table. The problem could arise from rounding, leading to multiple columns with the same name, and also from generating excessively long strings for column names.

Reply:
> Hello @brnovasco and @stuart-cls, thanks for advancing this. About the scale, METER is an appropriate unit for the M column. This column has the position readings of the stage encoder in meters. For example, for a typical mirror travel of 500 µm and 1024 pts, the step is ~0.49 µm. The M column will show 4.9E-7, therefore, meters.

Guys, just to make it clear, meters are the original unit of the data. However, converting it to microns would be more comfortable.